scrapy/pipelines/__init__.py

```diff
@@ -21,7 +21,8 @@ def _get_mwlist_from_settings(cls, settings):
                 category=ScrapyDeprecationWarning, stacklevel=1)
             # convert old ITEM_PIPELINE list to a dict with order 500
             item_pipelines = dict(zip(item_pipelines, range(500, 500+len(item_pipelines))))
-        return build_component_list(settings['ITEM_PIPELINES_BASE'], item_pipelines)
+            settings.set('ITEM_PIPELINES', item_pipelines)
```

Comment:
Good point. Providing a list still worked when the `and isinstance(value, Mapping)` was still present in `SettingsAttribute.set()`, but I also prefer dropping support for lists, especially as it will lead to unexpected behaviour for add-on developers.